### PR TITLE
⬆️ Update Kotlin to 1.9.10

### PR DESCRIPTION
**Pull request by kalinja**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```
Comment (smidf) [inline gradle/wrapper/gradle-wrapper.properties:3]: Is a new version necessary?

File: sample/build.gradle.kts [hunk 2]
```diff
@@ -90,6 +91,10 @@ sourceSets {
 
 tasks.withType<KotlinCompile> {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "17"
```
Comment (smidf) [inline sample/build.gradle.kts:94]: It shouldn't need a version update. Keep on the same version.

File: sample/build.gradle.kts [hunk 3]
```diff
@@ -90,6 +91,10 @@ sourceSets {
...
     }
 }
+
+tasks.withType<SwaggerCodeGenTask> {
+    project.tasks.findByName("kaptGenerateStubsKotlin")?.dependsOn(this)
```
Comment (smidf) [inline sample/build.gradle.kts:99]: Why do you create dependency on `kaptGenerateStubsKotlin`? I think it should depend on `KotlinCompile` so for example:
```kotlin
tasks.withType<KotlinCompile<*>> {
    dependsOn(/*name of tasks*/)
}
```

**Review by smidf — CHANGES_REQUESTED** [review]
Keep only changes related to version upgrades -> do not modify README and CHANGELOG files.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (kalinja) [reply gradle/wrapper/gradle-wrapper.properties:3]: Yes it was. As https://docs.gradle.org/current/userguide/compatibility.html

File: sample/build.gradle.kts (hunk 3 above)
Comment (kalinja) [reply sample/build.gradle.kts:99]: This is to resolve following issue when trying to build the sample.

A problem was found with the configuration of task ':sample:swagger-kotlin-petstore-def-and-opt-query' (type 'SwaggerCodeGenTask').
  - Gradle detected a problem with the following location: '/Users/kuba/Documents/work/projects/swagger-fork/sample/build/openapi/petstore-def-and-opt-query'.
    
    Reason: Task ':sample:kaptGenerateStubsKotlin' uses this output of task ':sample:swagger-kotlin-petstore-def-and-opt-query' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':sample:swagger-kotlin-petstore-def-and-opt-query' as an input of ':sample:kaptGenerateStubsKotlin'.
      2. Declare an explicit dependency on ':sample:swagger-kotlin-petstore-def-and-opt-query' from ':sample:kaptGenerateStubsKotlin' using Task#dependsOn.
      3. Declare an explicit dependency on ':sample:swagger-kotlin-petstore-def-and-opt-query' from ':sample:kaptGenerateStubsKotlin' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.4/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

File: sample/build.gradle.kts (hunk 2 above)
Comment (kalinja) [reply sample/build.gradle.kts]: I have downgraded it back to 1.8

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (smidf) [reply gradle/wrapper/gradle-wrapper.properties:3]: No, it isn't necessary :) In Gradle 8.4 is embedded Kotlin 1.9.10, but you can use Kotlin 1.9.10 with older versions of Gradle. But no problem, keep as is 👍

File: sample/build.gradle.kts (hunk 3 above)
Comment (smidf) [reply sample/build.gradle.kts:99]: Okey 👍

File: gradle.properties [hunk 4]
```diff
@@ -1,3 +1,3 @@
 org.gradle.jvmargs=-Xmx1536m
 
-version=2.3.0
+version=2.4.0
```
Comment (smidf) [inline gradle.properties:3]: Please, revert this change

File: README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # OpenApi 3 Codegen / Swagger
 
-[![Latest version](https://img.shields.io/github/v/release/eManPrague/swagger-codegen)](https://github.com/eManPrague/swagger-codegen/releases/tag/v2.3.0)
+[![Latest version](https://img.shields.io/github/v/release/eManPrague/swagger-codegen)](https://github.com/eManPrague/swagger-codegen/releases/tag/v2.4.0)
```
Comment (smidf) [inline README.md:3]: Please, revert this change

**Review by smidf — CHANGES_REQUESTED** [review]
